doc/concepts/labels.rst

```diff
@@ -385,6 +385,18 @@ Region expressions
 
       Branches 0 and 3, selected using ``(branch 0)`` and ``(branch 3)`` respectively.
 
+.. label:: (segment segment_id:integer)
+
+    Refer to a segment by its id. Note that segment ids depend on the construction
+    order of the morphology. While arbor's morphology loaders are stable in this regard,
+    it can become an issue when building morphologies by hand.
```

Comment:
How is this an issue when building by hand?
Segments are given ids in the order that they are added to the segment tree, and this is a strong guarantee.

Reply:
Hand-constructed `segment_trees` might be re-factored, producing the same morphology, but mixing up the labels.
Or is this a non-issue?

Reply:
I don't think that this is an issue.
This region expression is not intended for use by general users, because you are referring explicitly to a detail of a particular morphology, instead of a generic rule that can be applied to multiple morphologies.
It's intended use (and where it is used internally) is in software that constructs morphologies, which is expected to understand this.

E.g. the NeuroML segments do not correspond one-to-one with segments in Arbor, because a NeuroML segment can attached half way along another segment, in which case you need to split the parent segment into two Arbor segments. The NeuroML wrapper keeps track of the mapping between NeuroML segments and the Arbor segments that they are composed of, and presents them to the user using nice labels.

This could have a "hint" box that says that refering to segments directly is not best practice, and should be avoided because such rules can't be applied to generic morphologies.